app/models/notification.rb

```diff
@@ -0,0 +1,41 @@
+class Notification < ActiveRecord::Base
+  belongs_to :notification_type
+  belongs_to :initiator, :class_name => User, :foreign_key => 'user_id'
+  belongs_to :subject, :polymorphic => true
+  has_many :notification_recipients, :dependent => :delete_all
+  has_many :recipients, :class_name => User, :through => :notification_recipients, :source => :user
+
+  validates :notification_type, :presence => true
+  validates :initiator, :presence => true
+  validates :subject, :presence => true, :allow_nil => true
```

Comment:
the idea behind the subject is that you can refer to a specific obj, e.g. host X was deleted, in that case subject = host.

not every event has a subject (some subsystem might be down) hence the allow_nil, tbh I cant remember the line of thought around the two validations... :-\